lib/internal/fs/streams.js

```diff
@@ -339,7 +344,13 @@ WriteStream.prototype._write = function(data, encoding, cb) {
     });
   }
 
+  if (this.destroyed) return cb();
+
   fs.write(this.fd, data, 0, data.length, this.pos, (er, bytes) => {
+    // Return early if this stream has been destroyed. The close() call inside
+    // _destroy() may cause errors when writing and we don't want to emit those.
+    if (this.destroyed) return cb();
```

Comment:
Actually this part is a bit confusing for me. We shouldn't be swallowing the error here?

Reply:
We had a long conversation about swallowing errors after `destroy`here, #29197. The consensus was that we should not swallow until after `'close'`.

Reply:
I think we should fix `Writable` to wait calling `_destroy()` until the active write has completed.

The same workaround is however not possible for `_read`.

Reply:
Though I don't have a good alternative solution at the moment.